scripts/listauth-gates.py

```
@@ -0,0 +1,378 @@
+
+def screenshot(label):
+    out = round_dir / f"shot-{label}.png"
+    subprocess.run(["xcrun", "simctl", "io", args.udid, "screenshot", str(out)],
+                   capture_output=True)
+    return hashlib.sha256(out.read_bytes()).hexdigest() if out.exists() else None
+
+
+def type_input(text):
+    if args.no_input:
+        return False
+    typed = subprocess.run(["idb", "ui", "text", "--udid", args.udid, text],
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)